task/sql-tasks.js

```diff
@@ -280,7 +465,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+
+    let result = await db.query(`
+        SELECT DISTINCT
+            Customers.CompanyName,
+            Products.ProductName,
+            OrderDetails.UnitPrice AS 'PricePerItem'
+        FROM Customers
+        JOIN Orders ON Orders.CustomerID = Customers.CustomerID
+        JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+        JOIN Products ON Products.ProductID = OrderDetails.ProductID
+        WHERE OrderDetails.UnitPrice = (
+            SELECT 
+                MAX(OrderDetails.UnitPrice) FROM Customers AS customers2
+                JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+                JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+                WHERE customers2.CompanyName = Customers.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.